File: tracking-service/models/Tracking.js
const mongoose = require('mongoose');

const TrackingSchema = new mongoose.Schema({
  deliveryId: { type: String, required: true, index: true },
  driverId:   { type: String, required: true, index: true },
  location: {
    type: {
      type: String,
      enum: ['Point'],
      required: true
    },
    coordinates: {
      type: [Number], // [longitude, latitude]
      required: true
    }
  },
  timestamp: { type: Date, default: Date.now }
});

// Create 2dsphere index for geospatial queries
TrackingSchema.index({ location: '2dsphere' });

module.exports = mongoose.model('Tracking', TrackingSchema);

File: delivery_app/lib/models/tracking.dart
class Tracking {
  final String deliveryId;
  final String driverId;
  final double latitude;
  final double longitude;
  final DateTime timestamp;

  Tracking({
    required this.deliveryId,
    required this.driverId,
    required this.latitude,
    required this.longitude,
    required this.timestamp,
  });

  factory Tracking.fromJson(Map<String, dynamic> json) {
    return Tracking(
      deliveryId: json['deliveryId'],
      driverId: json['driverId'],
      latitude: json['latitude'],
      longitude: json['longitude'],
      timestamp: DateTime.parse(json['timestamp']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'deliveryId': deliveryId,
      'driverId': driverId,
      'latitude': latitude,
      'longitude': longitude,
      'timestamp': timestamp.toIso8601String(),
    };
  }
}
